mne/tests/test_epochs.py

```diff
@@ -1255,4 +1255,33 @@ def test_array_epochs():
     assert_equal(len(events), len(epochs.selection))
 
 
+def test_dataframe_time_indexing():
+    '''Make sure that conversion to dataframe
+    doesn't duplicate time indices due to rounding
+    error + floating point precision'''
+    # Synthesize data
+    data = np.random.randn(2, 10000)
+    data_event = np.zeros(data.shape[1])
+    data_event[[2, 2000]] = 1  # Insert some events
+    data_raw = np.vstack([data, data_event])
+
+    # For the Info object
+    ch_types = ['eeg']*data_raw.shape[0]
+    ch_names = ['{0}'.format(i) for i in range(data_raw.shape[0])]
+    ch_names[-1] = 'stim'
+    ch_types[-1] = 'misc'
+
+    # Create base MNE objects
+    info = create_info(ch_names, 1000, ch_types=ch_types)
+    raw = io.RawArray(data_raw, info)
+    events = find_events(raw, 'stim')
+    epochs = Epochs(raw, events, 1, -.2, 1.5)
```

Comment:
Maybe this is a dumb question, but why not just use `EpochsArray` directly?

Even if you don't do that, you can avoid adding the `stim` channel by passing the `events` array directly, saves an import and use of `find_events` as well